Sources/Parser.swift

```diff
@@ -13,8 +13,8 @@ public typealias SourceRange = CountableRange<SourceLocation>
 
 /// Input with location tracking
 public struct ParserInput {
-    public var lineStream: String.CharacterView
-    public var stream: String.CharacterView
+    public var lineStream: String
```

Comment:
`String` is very inefficient here. We don't want to make a copy every time we advance the input. Could you change it to `Substring`?

Reply:
Nevermind, I answered my own question.

Reply:
This note doesn't apply to parsing. Parsing needs to hold on to the entire string, *especially* because backtracking happens all the time. If you use `String`, you'll end up `malloc`ing every time you advance a character, let alone the cost of branches in the parse space and the fact that `String` is already slow. This is making what's already exponential (parser combinators) far slower.